1581086 update image filenames part two 09-30 (#14967)
dotnet/docs: docs/architecture/microservices/docker-application-development-process/docker-app-development-workflow.md

An application is composed of your own services plus additional libraries (dependencies). The following are the basic steps you usually take when building a Docker application, as illustrated in Figure 5-1.

> ![Diagram showing the 7 steps it takes to create a containerized app.](./media/docker-app-development-workflow/life-cycle-containerized-apps-docker-cli.png)

Reviewer: This would also be a good place to use the new image tag:

```markdown
:::image type="complex" source="<folderPath>" alt-text="<alt-text>":::
<long description here>
:::image-end:::
```

Using the original description as the long one in the tag.

The long description here gives context.

However, just because Visual Studio makes those steps automatic doesn't mean that you don't need to know what's going on underneath with Docker. Therefore, the following guidance details every step.

> ![Image of the Step 1 tile from the step-by-step workflow.](./media/docker-app-development-workflow/step-1-code-your-app.png)

Reviewer: Don't really think the alt-text here has much value, would just leave it mostly as it was.

```suggestion
![Image for Step 1.](./media/docker-app-development-workflow/step-1-code-your-app.png)
```

Also the title right below says it all, so no need to add anything (I missed that originally)

- **Visual Studio 2017** \
  [https://visualstudio.microsoft.com/downloads/](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2017)

> ![Image of the Step 2 tile from the step-by-step workflow.](./media/docker-app-development-workflow/step-2-write-dockerfile.png)

Reviewer: ```suggestion
![Image for Step 2.](./media/docker-app-development-workflow/step-2-write-dockerfile.png)
```

- **Create a base image**. Official Docker documentation.\
  <https://docs.docker.com/develop/develop-images/baseimages/>

> ![Image of the Step 3 tile from the step-by-step workflow.](./media/docker-app-development-workflow/step-3-create-dockerfile-defined-images.png)

Reviewer: ```suggestion
![Image for Step 3.](./media/docker-app-development-workflow/step-3-create-dockerfile-defined-images.png)
```

You can find the existing images in your local repository by using the docker images command, as shown in Figure 5-6.

![Console output from command docker images, showing existing images.](./media/docker-app-development-workflow/view-existing-images-with-docker-images.png)

**Figure 5-6.** Viewing existing images using the docker images command

### Creating Docker images with Visual Studio

When you use Visual Studio to create a project with Docker support, you don't explicitly create an image. Instead, the image is created for you when you press **F5** (or **Ctrl-F5**) to run the dockerized application or service. This step is automatic in Visual Studio and you won't see it happen, but it's important that you know what's going on underneath.

> ![Image of the optional Step 4 tile from the step-by-step workflow.](./media/docker-app-development-workflow/step-4-define-services-docker-compose-yml.png)

Reviewer: ```suggestion
![Image for the optional Step 4.](./media/docker-app-development-workflow/step-4-define-services-docker-compose-yml.png)
```

At the time of this writing, Visual Studio supports Docker Compose and Service Fabric orchestrators.

> ![Screenshot showing the Container Orchestrator Support option.](./media/docker-app-development-workflow/add-container-orchestrator-support-option.png)

Reviewer: ```suggestion
![Screenshot showing the Container Orchestrator Support option in the project context menu.](./media/docker-app-development-workflow/add-container-orchestrator-support-option.png)
```

**Figure 5-7**. Adding Docker support in Visual Studio 2017 by right-clicking an ASP.NET Core project

After you add orchestrator support to your solution in Visual Studio, you will also see a new node (in the `docker-compose.dcproj` project file) in Solution Explorer that contains the added docker-compose.yml files, as shown in Figure 5-8.

![Screenshot of docker-compose node in Solution Explorer.](./media/docker-app-development-workflow/docker-compose-tree-node.png)

**Figure 5-8**. The **docker-compose** tree node added in Visual Studio 2017 Solution Explorer

You could deploy a multi-container application with a single docker-compose.yml file by using the `docker-compose up` command. However, Visual Studio adds a group of them so you can override values depending on the environment (development or production) and execution type (release or debug). This capability will be explained in later sections.

> ![Image of the Step 5 tile from the step-by-step workflow.](./media/docker-app-development-workflow/step-5-run-containers-compose-app.png)

Reviewer: ```suggestion
![Image for the Step 5.](./media/docker-app-development-workflow/step-5-run-containers-compose-app.png)
```

The docker-compose up and docker run commands (or running and debugging the containers in Visual Studio) are adequate for testing containers in your development environment. But you should not use this approach for production deployments, where you should target orchestrators like [Kubernetes](https://kubernetes.io/) or [Service Fabric](https://azure.microsoft.com/services/service-fabric/). If you're using Kubernetes you have to use [pods](https://kubernetes.io/docs/concepts/workloads/pods/pod/) to organize containers and [services](https://kubernetes.io/docs/concepts/services-networking/service/) to network them. You also use [deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) to organize pod creation and modification.

> ![Image of the Step 6 tile from the step-by-step workflow.](./media/docker-app-development-workflow/step-6-test-app-microservices.png)

Reviewer: ```suggestion
![Image for the Step 6.](./media/docker-app-development-workflow/step-6-test-app-microservices.png)
```

## Step 6. Test your Docker application using your local Docker host

This step will vary depending on what your application is doing. In a simple .NET Core Web application that is deployed as a single container or service, you can access the service by opening a browser on the Docker host and navigating to that site, as shown in Figure 5-13. (If the configuration in the Dockerfile maps the container to a port on the host that is anything other than 80, include the host port in the URL.)

> ![Browser view of the response from localhost/API/values.](./media/docker-app-development-workflow/test-docker-app-locally-localhost.png)

Reviewer: Not much for a "screen", but just for consistency.

```suggestion
![Screenshot of the response from localhost/API/values.](./media/docker-app-development-workflow/test-docker-app-locally-localhost.png)
```

Effectively, the workflow when using Visual Studio is a lot simpler than if you use the editor/CLI approach. Most of the steps required by Docker related to the Dockerfile and docker-compose.yml files are hidden or simplified by Visual Studio, as shown in Figure 5-15.

> ![Diagram showing the five simplified steps it takes to create an app.](./media/docker-app-development-workflow/simplified-life-cycle-containerized-apps-docker-cli.png)

Reviewer: Please use extended image tag, as for 5-1.
